tests/test_provider_github.py

```diff
@@ -0,0 +1,316 @@
+import copy
+import unittest
+
+from keydra.providers import github
+
+from keydra.exceptions import ConfigException
+from keydra.exceptions import DistributionException
+from keydra.exceptions import RotationException
+
+from unittest.mock import call
+from unittest.mock import MagicMock
+from unittest.mock import patch
+
+
+GH_CREDS = {'username': 'user', 'password': 'pass'}
+
+
+IAM_SECRET = {
+    'key': 'secret_key',
+    'secret': 'secret_secret',
+    'provider': 'iam'
+}
+
+GH_DEST_ACCT = {
+    'key': 'secret_name',
+    'provider': 'github',
+    'source': 'key',
+    'scope': 'account',
+    'envs': ['*'],
+    'config': {'account_username': 'acct_user'}
+}
+
+GH_DEST_REPO = {
+    'key': 'secret_name',
+    'provider': 'github',
+    'source': 'key',
+    'scope': 'repository',
+    'config': {'repository': 'repo', 'account_username': 'acct_user'},
+    'envs': ['*']
+}
+
+GH_DEST_ENV = {
+    'key': 'secret_name',
+    'provider': 'github',
+    'source': 'key',
+    'scope': 'deployment',
+    'config': {
+        'repository': 'repo',
+        'environment': 'env',
+        'account_username': 'acct_user',
+        'create': True
+    },
+    'envs': ['*']
+}
+
+GH_DEST_REPO_NO_CREATE = {
```

Comment:
Doesn't look like we test (or have support) for `create` option in Github provider